File: trunk/cs241tp/tutorials/cache/tutorials/mv/ExerciseSolutions.cs
using System;
using System.Collections.Generic;
using System.Text;
using NUnit.Framework;
using InterSystems.Data.CacheTypes;
using InterSystems.Data.CacheClient;

namespace Exercises
{
    [TestFixture]
    public class ExerciseSolutions
    {

        CacheConnection cnCache;
        [SetUp]
        public void Connect()
        {

            string cacheString = "Server=localhost;" +
                                  "Port=1972;" +
                                  "Namespace=MYACCOUNT;" +
                                  "Password=SYS;" +
                                  "User ID=_system;";

            cnCache = new CacheConnection(cacheString);

        }
        [Test]
        public void AddPhone()
        {
            String number = "999-888-7777";
            cnCache.Open();
            String sql = "SELECT ItemId FROM MVFILE.PERSON WHERE NAME %Startswith 'SMITH'";
            CacheCommand command = new CacheCommand(sql, cnCache);
            CacheDataReader reader = command.ExecuteReader();

                while (reader.Read())
                {
                    String id = (String)reader["ItemId"];
                    Console.WriteLine("ItemId {0}", id);
                    MVFILE.PERSON person = MVFILE.PERSON.OpenId(cnCache, id);
                    person.PHONE.Add(number);
                    CacheStatus status = person.Save();
                    Console.WriteLine("Status {0}", status.Message);

                }


                cnCache.Close();

        }
        [Test]
        public void AddPerson()
        {
            cnCache.Open();
            String sql = "INSERT INTO MVFILE.PERSON (ItemId,NAME, HAIR, AGE, PHONE) VALUES ('15','WILLIAMS,ROGER', 'BRUNETTE','44','911-111-2222')";
            CacheCommand command = new CacheCommand(sql, cnCache);
            int rows = command.ExecuteNonQuery();
            Console.WriteLine("Updated rows: {0}",rows);
            cnCache.Close();

        }
        [Test]
        public void RunRoutine()
        {
            cnCache.Open();
           String search =  MVExercise.Update.SetValue(cnCache, "3", "HAIR", "1", "RED");
           cnCache.Close();

           Console.WriteLine("Search {0}", search);

        }

    }
}

File: trunk/cs241tp/tutorials/cache/tutorials/mv/ExerciseSolutions.java
/*
 * ExerciseSolutions.java
 *
 * Created on May 18, 2007, 12:00 PM
 *
 * To change this template, choose Tools | Template Manager
 * and open the template in the editor.
 */

package mvtutorial;
import MVFILE.*;
import java.sql.*;
import com.intersys.objects.*;
import java.util.*;
import java.sql.*;

public class ExerciseSolutions {
    static String url="jdbc:Cache://localhost:1972/MYACCOUNT";
    static Database db;
    static String newNumber = "999-888-7777";
  
    public static void connect() throws CacheException
    {
        
       db = CacheDatabase.getDatabase(url);
        
    }
    
    public static void addPerson() throws CacheException, SQLException
    {
        
            String sql = "INSERT INTO MVFILE.Person (ItemId,AGE,HAIR,NAME,PHONE) VALUES ('12','44','BRUNETTE','SMITH,ROGER','911-111-2222')";
            PreparedStatement pstmt = db.prepareStatement(sql);
            int status = pstmt.executeUpdate();
            System.out.printf("Status %s",status);
       
    }
    
    
    public static void addPhone()  throws CacheException
    {
            String sql="NAME %STARTSWITH ?";
            String[] qargs={"SMITH"};
            
          
           for (Iterator iter = db.openByQuery("MVFILE.PERSON",sql,qargs); iter.hasNext();)
           {
              MVFILE.PERSON person = (MVFILE.PERSON)iter.next();
              System.out.printf("Updatding %s",person.getNAME());
              person.getPHONE().add(newNumber);
              int status = person.save();
              System.out.printf("Status %s \n", status);                
           }
                 
        
    }
    
    public static void removePhone() throws CacheException
    {
         String sql="NAME %STARTSWITH ?";
            String[] qargs={"SMITH"};
            
            
           for (Iterator iter = db.openByQuery("MVFILE.PERSON",sql,qargs); iter.hasNext();)
           {
              MVFILE.PERSON person = (MVFILE.PERSON)iter.next();
              person.getPHONE().remove(newNumber);
              int status = person.save();
              System.out.printf("Status %s \n", status);                
           }
        
        
    }
    
    public static void RunRoutine(String id, String attr, String subValue, String value) throws CacheException
    {
        MVExercise.Update.SetValue(db,id,attr,subValue,value);
        
        
    }
    
    public static void main (String[] args)
    {
        try {
            ExerciseSolutions.connect();
            ExerciseSolutions.addPerson();
            ExerciseSolutions.addPhone();
            ExerciseSolutions.removePhone();
            
            
         ExerciseSolutions.RunRoutine("1","PHONE", "2", "999-999-9999");
            
        } catch (CacheException ex) {
            ex.printStackTrace();
        } 
       catch (SQLException ex){
            ex.printStackTrace();
        }
        finally{
            
            try{
            db.close();
            }
            catch (CacheException cex){}
        } 
              
    }
    
    
      
}
